### ISSUE TITLE
Error in versions newer than 1.7.0: useHref() may be used only in the context of a <Router> component

### ISSUE THREAD

**Issue by ecrvich** [remix-run/remix#4444]
### What version of Remix are you using?

1.7.4

### Steps to Reproduce

Here's my root component, largely copied from Remix examples:

```
export default function Root() {
	return (
		<html lang="en">
			<head>
				<Meta />
				<Links />
			</head>
			<body>
				<Sidenav />
				<Outlet />
				<ScrollRestoration />
				<Scripts />
				<LiveReload />
			</body>
		</html>
	);
}

```
Sidenav (which causes the error) looks like this:

```
export default function Sidenav() {
	return (
		<SideNav aria-label="Side navigation" isRail>
			<SideNavItems>
				{Object.keys(FunctionData).map((cardName) => (
					<Link key={cardName}
						to={cardName}>
						{FunctionData[cardName].cardTitle}
					</Link>
				))}
			</SideNavItems>
		</SideNav>
	);
}
```

SideNav (capital N) is an IBM Carbon UI component (similar to Google Material etc.), an implementation of a left-side navigation menu.

FYI, it fails even WITHOUT the Carbon wrapper, so Carbon is not the issue. For example, this fails the exact same way:

```
export default function Sidenav() {
	return (
		<>
			{Object.keys(FunctionData).map((cardName) => (
				<Link key={cardName}
					to={cardName}>
					{FunctionData[cardName].cardTitle}
				</Link>
			))}
		</>
	);
}
```

### Expected Behavior

Sidenav should render and the links should work. Remix 1.7.0 in fact does work fine with this implementation, and for now I'm prevented from updating to newer Remix versions until this is resolved. I'm open to code changes on my end of course, but haven't found any way yet to get it to work in 1.7.4...e.g., moving Sidenav into a sample child route that is rendered via Root's Outlet still gives the same error. The Sidenav component renders fine and the links work fine in 1.7.0, but not in 1.7.4 (pretty certain 1.7.2 and/or 1.7.3 similarly broke this, haven't tried 1.7.1).

### Actual Behavior

This is the error dump when I try to view any page of the app in 1.7.4 (no error and everything works great in 1.7.0):

```
Error: useHref() may be used only in the context of a <Router> component.
    at Object.invariant (C:\GitClones\ci-react-ts\node_modules\@remix-run\router\utils.ts:757:11)
    at Object.useHref (C:\GitClones\ci-react-ts\node_modules\react-router\lib\hooks.tsx:53:3)
    at Object.LinkWithRef [as render] (C:\GitClones\ci-react-ts\node_modules\react-router-dom\index.tsx:372:16)
    at ReactDOMServerRenderer.render (C:\GitClones\ci-react-ts\node_modules\react-dom\cjs\react-dom-server.node.development.js:3872:44)
    at ReactDOMServerRenderer.read (C:\GitClones\ci-react-ts\node_modules\react-dom\cjs\react-dom-server.node.development.js:3690:29)
    at renderToString (C:\GitClones\ci-react-ts\node_modules\react-dom\cjs\react-dom-server.node.development.js:4298:27)
    at handleRequest (C:\GitClones\ci-react-ts\build\index.js:1786:49)
    at handleDocumentRequest (C:\GitClones\ci-react-ts\node_modules\@remix-run\server-runtime\dist\server.js:431:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at requestHandler (C:\GitClones\ci-react-ts\node_modules\@remix-run\server-runtime\dist\server.js:72:18)
```

**Comment by machour** [comment]
@ecrvich could you create a little repository reproducing the issue? Hard to tell what's going wrong without that.

**Comment by ecrvich** [comment]
In trimming down the code for a simpler test case, I discovered the problem.

I had been using this import in Sidenav (and this works fine in Remix 1.7.0):

`import { Link } from "react-router-dom";`

I changed this to:

`import { Link } from "@remix-run/react";`

And everything now works fine with 1.7.4 as well.

I always try to use Remix imports over React Router imports where they exist in both places, but I guess I missed this one. And there are still some cases where components only seem to exist in React Router (like some hooks, I think), so it's seemingly a trial-and-error thing as to where to import from. No biggie. Sorry for the false alarm, feel free to close if you think that's appropriate!
